### PR TITLE
Add managed state to the TempoStack CRD

### LINKED ISSUE #411: Support a managed state like loki does
Its useful to debug things:
https://github.com/grafana/loki/blob/01f0ded7fcb57e3a7b26ffc1e8e3abf04a403825/operator/apis/loki/v1/lokistack_types.go#L13

cc @andreasgerstmayr

### PR DESCRIPTION
**Pull request by iblancasa**
Add support for managed state for Tempo instances.
Resolves #411.

### CODE REVIEW

File: internal/management/state/state.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package state
```
Comment (andreasgerstmayr) [inline internal/management/state/state.go:1]: This file looks like a duplicate of `controllers/tempo/internal/management/state/state.go`

File: .chloggen/add-managed-state.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. operator, github action)
+component: operator
+
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Add the `managementState` to the TempoStack spec. It enables/disables the reconciliation in the operator
```
Comment (pavolloffay) [inline .chloggen/add-managed-state.yaml:8]: `Add  `.spec.managedState` to `TempoStack`. It allows disabling reconciliation of the `TempoStack` CR.

File: controllers/tempo/internal/management/state/state_test.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+)
+
+var k8sClient client.Client
+var testEnv *envtest.Environment
+var testScheme *runtime.Scheme = scheme.Scheme
+
+func TestMain(m *testing.M) {
+	testEnv = &envtest.Environment{
+		CRDDirectoryPaths: []string{filepath.Join("..", "..", "..", "..", "..", "config", "crd", "bases")},
+		//WebhookInstallOptions: envtest.WebhookInstallOptions{
```
Comment (pavolloffay) [inline controllers/tempo/internal/management/state/state_test.go:31]: remove?

File: controllers/tempo/tempostack_controller.go [hunk 4]
```diff
@@ -74,6 +75,17 @@ type TempoStackReconciler struct {
 func (r *TempoStackReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	log := log.FromContext(ctx)
 	log = log.WithValues("tempo", req.NamespacedName)
+
+	managed, err := tempoStackState.IsManaged(ctx, req, r.Client)
```
[the remark this answers is not in the document]
Comment (pavolloffay) [reply controllers/tempo/tempostack_controller.go:79]: I would also prefer to have a single API call to get the operand and then act on it.

File: .chloggen/add-managed-state.yaml (hunk 2 above)
Comment (pavolloffay) [reply .chloggen/add-managed-state.yaml:8]: Let's document the behaviour of validation and defaulting webhook for unmanaged state.

File: controllers/tempo/tempostack_controller.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (iblancasa) [reply controllers/tempo/tempostack_controller.go:79]: Ok, it's OK for me. I just followed the same approach than in Loki

File: .chloggen/add-managed-state.yaml (hunk 2 above)
Comment (iblancasa) [reply .chloggen/add-managed-state.yaml:8]: @pavolloffay should I disable the validation and defaulting? That is not currently done

File: internal/management/state/state.go (hunk 1 above)
Comment (iblancasa) [reply internal/management/state/state.go:1]: I moved the file but... seems I didn't do properly in git. Thanks!

File: .chloggen/add-managed-state.yaml (hunk 2 above)
Comment (pavolloffay) [reply .chloggen/add-managed-state.yaml:8]: I think we should keep it enabled for now

**Review by andreasgerstmayr — APPROVED** [review]

**Pull request merged**